engine/escalationmanager/db.go

```diff
@@ -93,13 +100,15 @@ func NewDB(ctx context.Context, db *sql.DB, log *alertlog.Store) (*DB, error) {
 
 		newPolicies: p.P(`
 			with to_escalate as (
-				select alert_id, step.id ep_step_id, step.delay, step.escalation_policy_id, a.service_id
+				select alert_id, step.id ep_step_id, step.delay, step.escalation_policy_id, a.service_id, s.maintenance_expires_at
 				from escalation_policy_state state
 				join escalation_policy_steps step on
 					step.escalation_policy_id = state.escalation_policy_id and
 					step.step_number = 0
 				join alerts a on a.id = state.alert_id and (a.status = 'triggered' or state.force_escalation)
-				where state.last_escalation isnull
+				join services s on a.service_id = s.id
+				where state.last_escalation isnull and
+				(s.maintenance_expires_at isnull or s.maintenance_expires_at < now())
```

Comment:
Just do the `isnull` comparison here, cleanup will handle updating it, and we can simplify the work Postgres needs to do for the join.

```suggestion
				(s.maintenance_expires_at isnull)
```